connection.go

```diff
@@ -877,25 +965,10 @@ func (conn *Connection) fetchFuture(reqid uint32) (fut *Future) {
 func (conn *Connection) getFutureImp(reqid uint32, fetch bool) *Future {
 	shard := &conn.shard[reqid&(conn.opts.Concurrency-1)]
 	pos := (reqid / conn.opts.Concurrency) & (requestsMap - 1)
-	pair := &shard.requests[pos]
-	root := &pair.first
-	for {
-		fut := *root
-		if fut == nil {
-			return nil
-		}
-		if fut.requestId == reqid {
-			if fetch {
-				*root = fut.next
-				if fut.next == nil {
-					pair.last = root
-				} else {
-					fut.next = nil
-				}
-			}
-			return fut
-		}
-		root = &fut.next
+	if reqid%2 == 0 {
```

Comment:
i think it's worth to add a comment here about how's this work, because it is not obvious in any way.

Reply:
Isn't this logic a bit complicated? How did it came to this?

Reply:
Btw, is there any other request id logic except for equal comparison (and choosing the right shard)?

Reply:
> Isn't this logic a bit complicated? How did it came to this?

Without this logic, we had the common requestID counter and the problem was the next. When we receive a response from an instance, we fetch a request ID from IPROTO header and then we need to find the Future that is associated with this requestID. Then we need to search it in `requests` and `requestsWithCtx` lists both, cause the requestID can not be identified as the request with or without any context - that was the problem.

> Btw, is there any other request id logic except for equal comparison (and choosing the right shard)?
There are two sharding functions with request ID (for choosing shard and position in array of lists).

Reply:
Well, ok, let it be